### PR TITLE
bochs: refactor

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/bo/bochs/package.nix [hunk 1]
```diff
@@ -143,8 +143,10 @@ stdenv.mkDerivation (finalAttrs: {
       Intel x86 CPU, common I/O devices, and a custom BIOS.
     '';
     license = lib.licenses.lgpl2Plus;
+    mainProgram = "bochs";
     maintainers = with lib.maintainers; [ AndersonTorres ];
     platforms = lib.platforms.unix;
+    broken = stdenv.isDarwin;
```
Comment (wegank) [inline pkgs/by-name/bo/bochs/package.nix:149]: The package isn't broken on aarch64-darwin, however? https://hydra.nixos.org/job/nixpkgs/trunk/bochs.aarch64-darwin

Comment (AndersonTorres) [reply pkgs/by-name/bo/bochs/package.nix]: Still broken on Darwin ARM

Comment (Qyriad) [reply pkgs/by-name/bo/bochs/package.nix]: it's certainly still broken on aarch64-darwin for me
[z1xcamxmlhndf8vrzsk2gknrm4whqdvl-bochs-2.8.drv.log](https://github.com/NixOS/nixpkgs/files/15288092/z1xcamxmlhndf8vrzsk2gknrm4whqdvl-bochs-2.8.drv.log)

Comment (AndersonTorres) [reply pkgs/by-name/bo/bochs/package.nix]: OK, then I will break it on Darwin, and wait for the Darwin guys.

**Comment by AndersonTorres** [comment]
@wegank

File: pkgs/by-name/bo/bochs/package.nix [hunk 2]
```diff
@@ -60,80 +67,86 @@ stdenv.mkDerivation (finalAttrs: {
...
+
+    (lib.enableFeature (!stdenv.isDarwin) "e1000")
+    (lib.enableFeature (!stdenv.isDarwin) "es1370")
+    (lib.enableFeature (!stdenv.isDarwin) "ne2000")
+    (lib.enableFeature (!stdenv.isDarwin) "plugins")
+    (lib.enableFeature (!stdenv.isDarwin) "pnic")
+    (lib.enableFeature (!stdenv.isDarwin) "sb16")
   ];
 
+  outputs = [ "out" "lib" "doc" "man" ];
```
Comment (wegank) [inline pkgs/by-name/bo/bochs/package.nix:142]:
```suggestion
  outputs = [ "out" "doc" "man" ] ++ lib.optionals (!stdenv.isDarwin) [ "lib" ];
```

For unbreaking on aarch64-darwin.

**Pull request closed without merging**